workers/unity/Assets/Playground/Scripts/Metrics/MetricSendSystem.cs

```diff
@@ -36,54 +47,31 @@ protected override void OnUpdate()
                 return;
             }
 
-            timeElapsedSinceUpdate += Time.deltaTime;
-            
-            AddFpsSample();
-            if (timeElapsedSinceUpdate >= TimeBetweenMetricUpdatesSecs)
+            if (DateTime.Now.Ticks >= timeOfNextUpdate)
             {
-                timeElapsedSinceUpdate = 0;
-                var fps = CalculateFps();
-                var load = DefaultLoadCalculation(fps);
-                var metrics = new Improbable.Worker.Metrics
-                {
-                    Load = load
-                };
-                connection.SendMetrics(metrics);
-            }
-        }
+                var dynamicFps = CalculateFps();
+                Metrics.GaugeMetrics["Dynamic.FPS"] = dynamicFps;
+                Metrics.GaugeMetrics["Unity used heap size"] = GC.GetTotalMemory(false);
+                Metrics.Load = CalculateLoad(dynamicFps);
 
-        private static float DefaultLoadCalculation(float fps)
-        {
-            float targetFps = Application.targetFrameRate;
+                connection.SendMetrics(Metrics);
 
-            if (targetFps == -1)
-            {
-                targetFps = DefaultTargetFrameRate;
+                timeOfLastUpdate = DateTime.Now.Ticks;
+                timeOfNextUpdate = DateTime.Now.AddSeconds(TimeBetweenMetricUpdatesSecs).Ticks;
             }
-
-            return Mathf.Max(0.0f, (targetFps - fps) / (0.5f * targetFps));
         }
 
-        private void AddFpsSample()
+        private double CalculateLoad(double dynamicFps)
         {
-            if (fpsMeasurements.Count == MaxFpsSamples)
-            {
-                fpsMeasurements.Dequeue();
-            }
-
-            fpsMeasurements.Enqueue(1.0f / Time.deltaTime);
+            return Math.Max(0.0d, 0.5d * TargetFps / dynamicFps);
         }
 
-        private float CalculateFps()
+        private double CalculateFps()
         {
-            var fps = 0.0f;
-            foreach (var measurement in fpsMeasurements)
-            {
-                fps += measurement;
-            }
-
-            fps /= fpsMeasurements.Count;
-            return fps;
+            var elapsedTime = DateTime.Now.Ticks - timeOfLastUpdate;
+            var frames = Time.frameCount - lastFrameCount;
+            lastFrameCount = Time.frameCount;
+            return frames / TimeSpan.FromTicks(elapsedTime).TotalSeconds;
```

Comment:
If you store the `DateTime` objects you can do `return frames / (DateTime.Now - timeOfLastUpdate).TotalSeconds`

Reply:
will do as part of the move from ticks --> DateTime